### PR TITLE
Enhanced Pong

### PR DESCRIPTION
**Pull request by multimokia**
This finally got around to being finished, with overhauled dialogue and expressions from the original making it feel much more realistic and natural.

#3166 and related.

### CODE REVIEW

File: Monika After Story/game/pong.rpy [hunk 1]
```diff
@@ -429,40 +604,503 @@ init -1 python in mas_pong:
...
+    m 1eka "Thanks for playing Pong with me and letting me win."
+    m 1hua "I had a lot of fun! Let's play again sometime soon, okay?"
+
+    $ persistent._mas_pong_difficulty_change_next_game = PONG_DIFFICULTY_POWERDOWN;    
+    return
+
+#Post dlg Moni lose
+label mas_pong_dlg_loser_end:
+    m 1wuo "Wow, I was really trying that time."
+    m 1eua "you must have really been practicing to get so good."
```
Comment (MisterSimple) [inline Monika After Story/game/pong.rpy]:
```suggestion
    m 1eua "You must have really been practicing to get so good."
```

File: Monika After Story/game/pong.rpy [hunk 2]
```diff
@@ -429,40 +604,503 @@ init -1 python in mas_pong:
...
+    elif powerup_value_this_game == PONG_PONG_DIFFICULTY_POWERDOWNBIG:
+        m 1hua "I'm glad you won this time, [player]."
+
+    #Monika loses by a trickshot
+    elif pong_angle_last_shot >= 0.9 or pong_angle_last_shot <= -0.9:
+        if pong_monika_last_response_id == PONG_MONIKA_RESPONSE_LOSE_TRICKSHOT:
+            m 2wuo "[player]!"
+            m 2hksdlb "There's no way I could've hit that!"
+        else:
+            m 2wuo "Wow, there's no way I could hit that!"
```
Comment (MisterSimple) [inline Monika After Story/game/pong.rpy]:
```suggestion
            m 2wuo "Wow, there's no way I could've hit that!"
```

File: Monika After Story/game/pong.rpy [hunk 3]
```diff
@@ -429,40 +604,503 @@ init -1 python in mas_pong:
...
+                else:
+                    m 3hua "Looks like I won~"
+
+            $ pong_monika_last_response_id = PONG_MONIKA_RESPONSE_WIN_MEDIUM_GAME           
+
+        #Hard
+        elif pong_difficulty_before <= 15:
+            if pong_monika_last_response_id == PONG_MONIKA_RESPONSE_WIN_HARD_GAME:
+                m 1hub "Ahaha!"
+                m 2tsb "Am I playing too good for you?"
```
Comment (jmwall24) [inline Monika After Story/game/pong.rpy]:
```suggestion
                m 2tsb "Am I playing too well for you?"
```

File: Monika After Story/game/pong.rpy [hunk 4]
```diff
@@ -429,40 +604,503 @@ init -1 python in mas_pong:
...
+                    m 2eub "I win this round!"
+
+            $ pong_monika_last_response_id = PONG_MONIKA_RESPONSE_WIN_EXPERT_GAME
+
+        #Extreme
+        else:
+            if pong_monika_last_response_id == PONG_MONIKA_RESPONSE_WIN_EXTREME_GAME:
+                m 2duu "Not bad, [player]."
+                m 4eua "I gave it everything I had, so don't feel too bad for losing from time to time."
+                m "Keep practicing and you'll beat me"
```
Comment (jmwall24) [inline Monika After Story/game/pong.rpy]:
```suggestion
                m 4eub "Keep practicing and you'll beat me!"
```

File: Monika After Story/game/pong.rpy [hunk 5]
```diff
@@ -429,40 +604,503 @@ init -1 python in mas_pong:
...
+            else:
+                m 2hub "This time it's my win!"
```
Comment (jmwall24) [inline Monika After Story/game/pong.rpy]:
```suggestion
                m 2hub "This time, the win is mine!"
```

File: Monika After Story/game/pong.rpy [hunk 6]
```diff
@@ -429,40 +604,503 @@ init -1 python in mas_pong:
...
+        $ pong_monika_last_response_id = PONG_MONIKA_RESPONSE_LOSE_WITHOUT_HITTING_BALL
+
+    #Player starts playing seriously and wins after losing at least 3 times on purpose
+    elif instant_loss_streak_counter_before >= 3 and persistent._mas_pm_ever_let_monika_win_on_purpose:
+        m 2tsu "So you're playing seriously now?"
+        m 2tfu "Let's find out how good you really are, [player]!"
+
+    #Player wins after losing at least three times in a row
+    elif loss_streak_counter_before >= 3:
+        m 4eub "Congrats [player]!"
```
Comment (jmwall24) [inline Monika After Story/game/pong.rpy]:
```suggestion
        m 4eub "Congrats, [player]!"
```

File: Monika After Story/game/pong.rpy [hunk 7]
```diff
@@ -429,40 +604,503 @@ init -1 python in mas_pong:
...
+                else:
+                    m 2hua "You won! Congratulations!"
+
+            $ pong_monika_last_response_id = PONG_MONIKA_RESPONSE_LOSE_HARD_GAME
+
+        #Expert
+        elif pong_difficulty_before <= 20:
+            if pong_monika_last_response_id == PONG_MONIKA_RESPONSE_LOSE_EXPERT_GAME:
+                m 2wuo "Wow,{w=1} I'm really trying...{w=1}you're unstoppable!"
+                m 2tfu "But I'm sure I'll beat you sooner or later [player]."
```
Comment (jmwall24) [inline Monika After Story/game/pong.rpy]:
```suggestion
                m 2tfu "But I'm sure I'll beat you sooner or later, [player]."
```

**Comment by ThePotatoGuy** [comment]
@multimokia whats the major changes in terms of gameplay? does it feel more fair, realistic, more fun than before?

File: Monika After Story/game/pong.rpy [hunk 8]
```diff
@@ -429,40 +604,503 @@ init -1 python in mas_pong:
...
+    elif win_streak_counter == 5:
+        m 2wud "[player]..."
+        m 2tsu "Have you been practicing?"
+        m 3hksdlb "I don't know what happened but I don't stand a chance against you!"
+        m 1eka "Could you go a little bit easier on me please?"
+        m 3hub "I would really appreciate it~"
+        $ monika_asks_to_go_easy = True
+
+    #Monika loses a long game
+    elif ball_paddle_bounces > 10 and ball_paddle_bounces > pong_difficulty_before * 1/2:
```
Comment (ThePotatoGuy) [inline Monika After Story/game/pong.rpy]: so I am assuming this should be 1/2 (0.5) times `pong_difficulty_before`, correct?
If so, just use 0.5 as integer precision here probably isn't needed.

File: Monika After Story/game/pong.rpy [hunk 9]
```diff
@@ -429,40 +604,503 @@ init -1 python in mas_pong:
...
+    #Monika wins a second time after the player got a 3+ winstreak
+    elif pong_monika_last_response_id == PONG_MONIKA_RESPONSE_WIN_AFTER_PLAYER_WON_MIN_THREE_TIMES:
+        m 1hua "Ehehe!"
+        m 1tub "Keep up, [player]!"
+        m 2tfu "It looks like your streak is over!"
+
+        $ pong_monika_last_response_id = PONG_MONIKA_RESPONSE_SECOND_WIN_AFTER_PLAYER_WON_MIN_THREE_TIMES
+
+    #Monika wins a long game
+    elif ball_paddle_bounces > 9 and ball_paddle_bounces > pong_difficulty_before * 1/2:
```
Comment (ThePotatoGuy) [inline Monika After Story/game/pong.rpy]: another spot with 1/2

**Comment by multimokia** [comment]
Overall, pong feels more realistic, where Monika's difficulty level changes based on how you play, plus the dialogue feels more like how it would as if two people were playing together, instead of it being an almost robotic loop. I'd say it's more fun, since it actually feels like you're playing with someone now.

**Comment by Rai99** [comment]
Presumably, with 100% less T piece tetris blocks.

**Pull request merged**